WordPress/src/main/java/org/wordpress/android/ui/reader/usecases/ReaderSiteFollowUseCase.kt

```diff
@@ -83,4 +97,24 @@ class ReaderSiteFollowUseCase @Inject constructor(
             object RequestFailed : Failed()
         }
     }
+
+    sealed class Param {
+        abstract val blogId: Long
+        abstract val feedId: Long
+        abstract val blogName: String
+        data class FromPost(
+            val post: ReaderPost
+        ) : Param() {
+            override val blogId: Long = post.blogId
+            override val feedId: Long = post.feedId
+            override val blogName: String = post.blogName
+        }
+
+        data class RecommendedSite(
+            override val blogId: Long,
+            override val feedId: Long,
+            override val blogName: String,
+            val isAskingToFollow: Boolean
+        ) : Param()
+    }
```

Comment:
In order to reuse this use case, I decided to create a sealed class for a param object that is passed to the toggleFollow function.
This allows us to pass different properties and based on the type map it to the correct data.

Reply:
I simplified this, so no need for sealed class anymore